lib/jsdom/living/nodes/HTMLLabelElement-impl.js

```diff
@@ -48,6 +48,26 @@ function sendClickToAssociatedNode(node) {
 }
 
 class HTMLLabelElementImpl extends HTMLElementImpl {
+  get control() {
+    if (this.hasAttribute("for")) {
+      const node = this.ownerDocument.getElementById(this.getAttribute("for"));
+      if (node && isLabelable(node)) {
+        return node;
+      }
+      return undefined;
```

Comment:
This should probably be null.

Reply:
Fixed. Also changed the activation behaviour to make use of this. it didn't handle the _wrong-for-attribute-case_.